File: RadVolontera/RadVolontera.Models/CompanyEvent/CompanyEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadVolontera.Models.CompanyEvent
{
    public  class CompanyEvent
    {
        public long Id { get; set; }
        public long CompanyId { get; set; }
        public Company.Company Company { get; set; }
        public string EventName { get; set; }
        public DateTime EventDate { get; set; }
        public string Location { get; set; }
        public string Time { get; set; }
        public bool Registered { get; set; }
    }
}

File: RadVolontera/UI/radvolontera_admin/lib/models/compnay_event/company_event.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:radvolontera_admin/models/company/company.dart';

part 'company_event.g.dart';

@JsonSerializable()
class CompanyEventModel {
  int? id;
  String? eventName;
  String location;
  String? time;
 int? companyId;
 CompanyModel? company;
 DateTime? eventDate;
 int price;
CompanyEventModel(this.id, this.eventName, this.location, this.time, this.company, this.companyId, this.eventDate,this.price);
      /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory CompanyEventModel.fromJson(Map<String, dynamic> json) => _$CompanyEventModelFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$CompanyEventModelToJson(this);
}
